numba/inline_closurecall.py

```diff
@@ -133,14 +133,17 @@ def _inline_reduction(self, work_list, block, i, expr, call_name):
         require(not self.parallel_options.reduction)
         require(call_name == ('reduce', 'builtins') or
                 call_name == ('reduce', '_functools'))
-        if len(expr.args) != 3:
+        if len(expr.args) not in (2, 3):
             raise TypeError("invalid reduce call, "
                             "three arguments including initial "
```

Comment:
This error might need touching up, since we now also accept 2 args?

Reply:
Indeed. WIP :)

Reply:
It's also a bit segfaulty at present!